native/src/main/java/io/ballerina/stdlib/grpc/Message.java

```diff
@@ -714,34 +714,39 @@ private String extractMessageNameWithoutNamespace(String messageName) {
     }
 
-    private com.google.protobuf.Descriptors.Descriptor getDescriptorFromRecord(RecordType recordType,
-                                                                               String messageName)
+    private Descriptors.Descriptor getDescriptorFromRecord(RecordType recordType, String messageName)
             throws InvalidProtocolBufferException {
-
         if (isDescriptorAnnotationAvailable(recordType)) {
-            BMap<BString, Object> annotations = recordType.getAnnotations();
-            String annotation = filterDescAnnotation(annotations);
-            byte[] annotationAsBytes = ServicesBuilderUtils.hexStringToByteArray(annotation);
-            DescriptorProtos.FileDescriptorProto file = DescriptorProtos.FileDescriptorProto
-                    .parseFrom(annotationAsBytes);
-            Descriptors.FileDescriptor fileDescriptor;
-            try {
-                fileDescriptor = Descriptors.FileDescriptor.buildFrom(file, new Descriptors.FileDescriptor[]{}, true);
-                Descriptors.Descriptor desc = fileDescriptor.findMessageTypeByName(recordType.getName());
-                if (desc != null) {
-                    return desc;
-                }
-            } catch (Descriptors.DescriptorValidationException e) {
-                return MessageRegistry.getInstance().getMessageDescriptor(messageName);
-            }
+            return extractDescriptorFromAnnotation(recordType.getAnnotations(), recordType.getName());
         }
-        return MessageRegistry.getInstance().getMessageDescriptor(messageName);
+        // When the user returns an anonymous record as a stream, annotation values are not included in the record type
+        RecordType createdRecordType = (RecordType) ValueCreator.createRecordValue(recordType.getPackage(),
+                messageName).getType();
+        if (isDescriptorAnnotationAvailable(createdRecordType)) {
+            return extractDescriptorFromAnnotation(createdRecordType.getAnnotations(), this.messageName);
+        }
+        return getMessageDescriptorFromMessageRegistry(messageName);
+    }
+
+    private Descriptors.Descriptor extractDescriptorFromAnnotation(BMap<BString, Object> annotations, String msgName)
+            throws InvalidProtocolBufferException {
+        String annotation = filterDescAnnotation(annotations);
+        byte[] annotationAsBytes = ServicesBuilderUtils.hexStringToByteArray(annotation);
+        DescriptorProtos.FileDescriptorProto file = DescriptorProtos.FileDescriptorProto.parseFrom(annotationAsBytes);
+        try {
+            Descriptors.FileDescriptor fileDescriptor = Descriptors.FileDescriptor
+                    .buildFrom(file, new Descriptors.FileDescriptor[]{}, true);
+            Descriptors.Descriptor desc = fileDescriptor.findMessageTypeByName(msgName);
+            if (desc != null) {
+                return desc;
+            }
+        } catch (Descriptors.DescriptorValidationException e) { }
```

Comment:
Shall we add a comment here, explaining why we ignore the exception?

Reply:
Added in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/106ae179a53549ba2f9358169b4692daaf4a0c57